src/vanilla/Model/CompositeTypeTS.cs

```diff
@@ -33,6 +33,35 @@ public override Property Add(Property item)
             return result;
         }
 
+        private List<CompositeTypeTS> polymorphicUnionCases;
```

Comment:
Why is this a List? Is it going to be modified after it is initially created?

Reply:
I'd like for it to be immutable, computed lazily on the first get and then cached. This is used in `ReferenceName` to decide whether to refer to the type as `FooUnion` or just `Foo` based on whether any polymorphic subtypes exist.

Reply:
I get that, but why is it a List and not an Array or an IEnumerable?

Reply:
Not IEnumerable because I want it to be unambiguous that "work" isn't being done upon enumeration.

Reply:
If that's the case, then why not an array? Also, why does the laziness of the work matter to you? Isn't your only requirement that it be a sequence of values?

Reply:
It can be an array. We can convey even more that you're not meant to overwrite values in the array by using `IReadOnlyList`. I just want to avoid repeated execution of the query through multiple usages of `ReferenceName`.